### ISSUE TITLE
Issues with snap package

### ISSUE THREAD

**Issue by thc202** [zaproxy/zaproxy#4955]
Tracker issue for issues found while using snap package (#4922).

Using:
```
Name     Version    Rev   Tracking  Publisher  Notes
zaproxy  2.7.0      2     beta      psiinon    devmode
```
(installed with `snap install --devmode --beta zaproxy`)

The JxBrowser is not starting:
```
03:13:16 SEVERE: There are next missing dependencies: 
	browsercore64 => libgconf-2.so.4, libXss.so.1, libgtk-3.so.0, libgdk-3.so.0
	libbrowsercore64.so => libgconf-2.so.4, libXss.so.1, libgtk-3.so.0, libgdk-3.so.0
03:13:16 SEVERE: Failed to start IPC process.
java.lang.IllegalStateException: Missing dependendecies have been detected. Check the log for details.
	at com.teamdev.jxbrowser.chromium.internal.ipc.ExternalChromiumProcessLinux.preProcessRun(SourceFile:3156)
	at com.teamdev.jxbrowser.chromium.internal.ipc.ExternalChromiumProcess.doStart(SourceFile:62)
	at com.teamdev.jxbrowser.chromium.internal.ipc.ChromiumProcess.start(SourceFile:202)
	at com.teamdev.jxbrowser.chromium.internal.ipc.d.run(SourceFile:196)
	at java.lang.Thread.run(Thread.java:748)
37825 [AWT-EventQueue-0] ERROR org.zaproxy.zap.extension.jxbrowserlinux64.ExtensionJxBrowserLinux64  - Failed to start IPC process.
com.teamdev.jxbrowser.chromium.internal.ipc.IPCException: Failed to start IPC process.
	at com.teamdev.jxbrowser.chromium.internal.ipc.d.run(SourceFile:203)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalStateException: Missing dependendecies have been detected. Check the log for details.
	at com.teamdev.jxbrowser.chromium.internal.ipc.ExternalChromiumProcessLinux.preProcessRun(SourceFile:3156)
	at com.teamdev.jxbrowser.chromium.internal.ipc.ExternalChromiumProcess.doStart(SourceFile:62)
	at com.teamdev.jxbrowser.chromium.internal.ipc.ChromiumProcess.start(SourceFile:202)
	at com.teamdev.jxbrowser.chromium.internal.ipc.d.run(SourceFile:196)
	... 1 more
```

Firefox is not found (even if the binary is manually specified), related to snap permissions?
```
59217 [Thread-11] ERROR org.zaproxy.zap.extension.quickstart.launch.ExtensionQuickStartLaunch  - Cannot find firefox binary in PATH. Make sure firefox is installed. OS appears to be: LINUX
Build info: version: 'unknown', revision: 'unknown', time: 'unknown'
System info: host: 'mars', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.15.0-33-generic', java.version: '1.8.0_181'
Driver info: driver.version: FirefoxDriver
org.openqa.selenium.WebDriverException: Cannot find firefox binary in PATH. Make sure firefox is installed. OS appears to be: LINUX
Build info: version: 'unknown', revision: 'unknown', time: 'unknown'
System info: host: 'mars', ip: '127.0.1.1', os.name: 'Linux', os.arch: 'amd64', os.version: '4.15.0-33-generic', java.version: '1.8.0_181'
Driver info: driver.version: FirefoxDriver
	at org.openqa.selenium.firefox.FirefoxBinary.<init>(FirefoxBinary.java:116)
	at org.openqa.selenium.firefox.FirefoxOptions.getBinary(FirefoxOptions.java:218)
	at org.openqa.selenium.firefox.FirefoxDriver.toExecutor(FirefoxDriver.java:155)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:120)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getWebDriverImpl(ExtensionSelenium.java:683)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getWebDriver(ExtensionSelenium.java:628)
	at org.zaproxy.zap.extension.selenium.internal.BuiltInSingleWebDriverProvider.getWebDriver(BuiltInSingleWebDriverProvider.java:62)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getWebDriverImpl(ExtensionSelenium.java:600)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getWebDriver(ExtensionSelenium.java:491)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getProxiedBrowser(ExtensionSelenium.java:571)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getProxiedBrowserByName(ExtensionSelenium.java:540)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getProxiedBrowserByName(ExtensionSelenium.java:528)
	at org.zaproxy.zap.extension.selenium.ExtensionSelenium.getProxiedBrowserByName(ExtensionSelenium.java:519)
	at org.zaproxy.zap.extension.quickstart.launch.ExtensionQuickStartLaunch$2$1.run(ExtensionQuickStartLaunch.java:278)
	at java.lang.Thread.run(Thread.java:748)
```

A file containing `çã` breaks the choose file dialogue:
```
406759 [Basic L&F File Loading Thread] ERROR org.zaproxy.zap.ZAP$UncaughtExceptionLogger  - Exception in thread "Basic L&F File Loading Thread"
java.nio.file.InvalidPathException: Malformed input or input contains unmappable characters: /path/to/file/????
	at sun.nio.fs.UnixPath.encode(UnixPath.java:147)
	at sun.nio.fs.UnixPath.<init>(UnixPath.java:71)
	at sun.nio.fs.UnixFileSystem.getPath(UnixFileSystem.java:281)
	at java.nio.file.Paths.get(Paths.java:84)
	at sun.awt.shell.ShellFolder.getShellFolder(ShellFolder.java:247)
	at javax.swing.filechooser.FileSystemView.getFiles(FileSystemView.java:490)
	at javax.swing.plaf.basic.BasicDirectoryModel$LoadFilesThread.run0(BasicDirectoryModel.java:239)
	at javax.swing.plaf.basic.BasicDirectoryModel$LoadFilesThread.run(BasicDirectoryModel.java:228
```

**Comment by thc202** [comment]
Revise after #4965.

**Comment by thc202** [comment]
Using the snap built from #4965 fixed the Firefox launching issue. JxBrowser fails with a different error:
```
03:27:03 SEVERE: [0926/152703.446097:ERROR:gl_context_glx.cc(186)] Failed to create GL context with glXCreateContextAttribsARB.
03:27:03 SEVERE: [0926/152703.446952:ERROR:gpu_info_collector.cc(51)] gl::init::CreateGLContext failed
03:27:03 SEVERE: [0926/152703.446965:ERROR:gpu_info_collector.cc(119)] Could not create context for info collection.
03:27:03 SEVERE: [0926/152703.446987:ERROR:gpu_init.cc(94)] gpu::CollectGraphicsInfo failed (fatal).
03:27:03 SEVERE: [0926/152703.452217:ERROR:gpu_child_thread.cc(264)] Exiting GPU process due to errors during initialization
03:27:03 SEVERE: [0926/152703.535142:ERROR:browser_gpu_channel_host_factory.cc(103)] Failed to launch GPU process.
151087 [AWT-EventQueue-0] ERROR org.zaproxy.zap.extension.jxbrowserlinux64.ExtensionJxBrowserLinux64  - Failed to initialize IPC. Remote process doesn't respond.
com.teamdev.jxbrowser.chromium.internal.ipc.IPCException: Failed to initialize IPC. Remote process doesn't respond.
	at com.teamdev.jxbrowser.chromium.internal.ipc.IPC.a(SourceFile:175)
	at com.teamdev.jxbrowser.chromium.internal.ipc.IPC.start(SourceFile:144)
	at com.teamdev.jxbrowser.chromium.Browser.<init>(SourceFile:398)
	at com.teamdev.jxbrowser.chromium.Browser.<init>(SourceFile:351)
	at com.teamdev.jxbrowser.chromium.Browser.<init>(SourceFile:312)
	at org.zaproxy.zap.extension.jxbrowser.ZapBrowserPanel.getBrowser(ZapBrowserPanel.java:86)
	at org.zaproxy.zap.extension.jxbrowser.BrowserPanel.<init>(BrowserPanel.java:75)
	at org.zaproxy.zap.extension.jxbrowser.BrowserPanel.<init>(BrowserPanel.java:67)
	at org.zaproxy.zap.extension.jxbrowser.ZapBrowserPanel.<init>(ZapBrowserPanel.java:65)
	at org.zaproxy.zap.extension.jxbrowser.ZapBrowserFrame.getNewBrowserPanel(ZapBrowserFrame.java:58)
	at org.zaproxy.zap.extension.jxbrowser.BrowserFrame.addNewBrowserPanel(BrowserFrame.java:210)
	at org.zaproxy.zap.extension.jxbrowser.BrowserFrame.<init>(BrowserFrame.java:83)
	at org.zaproxy.zap.extension.jxbrowser.BrowserFrame.<init>(BrowserFrame.java:68)
	at org.zaproxy.zap.extension.jxbrowser.BrowserFrame.<init>(BrowserFrame.java:64)
	at org.zaproxy.zap.extension.jxbrowser.ZapBrowserFrame.<init>(ZapBrowserFrame.java:45)
	at org.zaproxy.zap.extension.jxbrowserlinux64.ExtensionJxBrowserLinux64$3.run(ExtensionJxBrowserLinux64.java:106)
	at java.awt.event.InvocationEvent.dispatch(InvocationEvent.java:311)
	at java.awt.EventQueue.dispatchEventImpl(EventQueue.java:758)
	at java.awt.EventQueue.access$500(EventQueue.java:97)
	at java.awt.EventQueue$3.run(EventQueue.java:709)
	at java.awt.EventQueue$3.run(EventQueue.java:703)
	at java.security.AccessController.doPrivileged(Native Method)
	at java.security.ProtectionDomain$JavaSecurityAccessImpl.doIntersectionPrivilege(ProtectionDomain.java:74)
	at java.awt.EventQueue.dispatchEvent(EventQueue.java:728)
	at java.awt.EventDispatchThread.pumpOneEventForFilters(EventDispatchThread.java:205)
	at java.awt.EventDispatchThread.pumpEventsForFilter(EventDispatchThread.java:116)
	at java.awt.EventDispatchThread.pumpEventsForHierarchy(EventDispatchThread.java:105)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:101)
	at java.awt.EventDispatchThread.pumpEvents(EventDispatchThread.java:93)
	at java.awt.EventDispatchThread.run(EventDispatchThread.java:82)
```

The choose file dialogue fails with the same error.

**Comment by Holston5** [comment]
From above:

The snapdoesn't work, no previous versions of ZAP are installed. OpendJDK8 IS installed (though not as a snap package) and is known to work fine. Awk is also installed, though not as a snap package.

zaproxy
/snap/zaproxy/2/zap.sh: line 46: awk: command not found
/snap/zaproxy/2/zap.sh: line 48: awk: command not found
/snap/zaproxy/2/zap.sh: line 51: [: -ge: unary operator expected
/snap/zaproxy/2/zap.sh: line 54: [: -ge: unary operator expected
Exiting: ZAP requires a minimum of Java 8 to run, found

Steps to reproduce the behavior:

    snap install --beta --devmode zaproxy
    zaproxy

whereis java: /usr/bin/java /usr/share/java
whereis awk: /usr/bin/awk /usr/share/awk
OS: Solus 3.9999

**Comment by thc202** [comment]
@Holston5 would you mind trying the latest version? https://github.com/zaproxy/zaproxy/wiki/Downloads#snapcraft

**Comment by kravietz** [comment]
Getting the same symptom - JxBrowser not starting but different error:
```
12:51:57 SEVERE: There are next missing dependencies: 
	browsercore64 => libgconf-2.so.4
	libbrowsercore64.so => libgconf-2.so.4
12:51:57 SEVERE: Failed to start IPC process.
java.lang.IllegalStateException: Missing dependendecies have been detected. Check the log for details.
```
Zaproxy 2.7.0 from snap, Ubuntu 18.10

**Comment by thc202** [comment]
> A file containing çã breaks the choose file dialogue:

This is fixed in the latest snap (2.10.0).

We are no longer providing JxBrowser, so that's no longer an issue as well.

Leaving open in case someone finds other issues.
